Cargo.toml

```diff
@@ -0,0 +1,8 @@
+[package]
```

Comment:
Here you want to use a workspace instead. That allows for sharing a bunch of things between all crates that are members of the workspace, including external crates dependencies. See https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table and https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html for more details. 

Then all other crates are member of that workspace

Reply:
We have chosen to not use workspace. Considering that both the agent and initramfs are completely separate projects (will not directly interact with api & lumper), the choice of not using a workspace seems to us the most judicious.

Reply:
> We have chosen to not use workspace. Considering that both the agent and initramfs are completely separate projects (will not directly interact with api & lumper), the choice of not using a workspace seems to us the most judicious.

Workspaces do not necessarily contain logically dependant crates. They will share the same output directory, and most importantly the same set of dependent crates. That's quite useful for avoiding dependencies duplication, conflict and making them easier to maintain.